src/workerd/api/gpu/gpu-buffer.h

```diff
@@ -36,17 +40,26 @@ class GPUBuffer : public jsg::Object {
     inline bool Intersects(uint64_t s, uint64_t e) const {
       return s < end && e > start;
     }
-    kj::ArrayPtr<kj::byte> buffer;
+    std::unique_ptr<jsg::V8Ref<v8::ArrayBuffer>> buffer;
```

Comment:
A `std::unique_ptr` holding a `jsg::V8Ref` is super odd and unnecessary. The `jsg::V8Ref` is itself an owning smart ptr. What is the intended purpose here?

Reply:
I went a bit crazy with the types, fixed.